src/headless/yare_impl/headless.rs

```diff
@@ -157,11 +626,143 @@ impl<F: Fn(u32)> Headless<F> {
         None
     }
 
-    pub fn simulate(mut self) -> Outcome {
+    pub fn tick(&mut self) -> Option<Outcome> {
+        self.update_env();
+
+        for player_index in 0..self.players.len() {
+            self.gather_commands(player_index)
+        }
+        self.process_commands()
+    }
+
+    pub fn simulate(mut self) -> SimulationResult {
         loop {
             if let Some(outcome) = self.tick() {
-                return outcome;
+                self.write_replay();
```

Comment:
Maybe make writing the replay optional?

Reply:
```rust
pub fn write_replay(&self) {
        if let Some(path) = &self.replay_path {
            let replay = serde_json::to_string(&self.replay).unwrap();
            let mut file = File::create(path).unwrap();
            file.write_all(replay.as_bytes()).unwrap();
        }
    }
```
Replays are only written if you supply a file path to write them to. If you pass None instead when you make the headless it doesn't write a replay